### PR TITLE
40 in memory data store

### LINKED ISSUE #40: Implement in memory data storage
Implement a version of the Forum storage interface so that an initial version of the app can function on a connected local mesh, up until a distributed storage solution can be implemented.

### PR DESCRIPTION
**Pull request by MarshallAsch**
closes #40 . Basic implementation of a local data store has been completed, Note that it is not 100% finished, the function `updateUser(DhtProto.User user);`  has not been implemented yet and is left as a stub that returns `false`. This should allow most of the app functionality to be mocked on a single device then adding in RM to share between devices should be easy after that.

### CODE REVIEW

File: app/src/main/java/ca/marshallasch/veil/MemoryStore.java [hunk 1]
```diff
@@ -0,0 +1,521 @@
...
+import java.util.Locale;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import ca.marshallasch.veil.exceptions.TooManyResultsException;
+import ca.marshallasch.veil.proto.DhtProto;
+import ca.marshallasch.veil.utilities.Util;
+
+import static ca.marshallasch.veil.proto.DhtProto.KeywordType.*;
+
+/**
```
Comment (weihanli101) [inline app/src/main/java/ca/marshallasch/veil/MemoryStore.java]: Add in overview for what the class is for.

File: app/src/main/java/ca/marshallasch/veil/MemoryStore.java [hunk 2]
```diff
@@ -0,0 +1,521 @@
...
+        String hash = Util.generateHash(comment.toByteArray());
+
+        DhtProto.DhtWrapper wrapper = DhtProto.DhtWrapper.newBuilder()
+                .setComment(comment)
+                .setType(DhtProto.MessageType.COMMENT)
+                .build();
+
+        insert(wrapper, hash);
+
+        // the keyword item is empty
```
Comment (weihanli101) [inline app/src/main/java/ca/marshallasch/veil/MemoryStore.java]: Make clear what this item is. Aka the search item for the comment you are about to insert.

File: app/src/main/java/ca/marshallasch/veil/MemoryStore.java (hunk 1 above)
Comment (MarshallAsch) [reply app/src/main/java/ca/marshallasch/veil/MemoryStore.java]: why comment import lines?

File: app/src/main/java/ca/marshallasch/veil/MemoryStore.java (hunk 2 above)
Comment (MarshallAsch) [reply app/src/main/java/ca/marshallasch/veil/MemoryStore.java]: Fixed in commit e4bdcda

File: app/src/main/java/ca/marshallasch/veil/MemoryStore.java (hunk 1 above)
Comment (weihanli101) [reply app/src/main/java/ca/marshallasch/veil/MemoryStore.java]: It is meant to be addressed at the top of your javadoc comment. Just add in a description (overview) of what this class is used for and the content it holds.

Comment (MarshallAsch) [reply app/src/main/java/ca/marshallasch/veil/MemoryStore.java]: but you could just look at the class, that particular ENUM is not anything special.

Comment (MarshallAsch) [reply app/src/main/java/ca/marshallasch/veil/MemoryStore.java]: Fixed in commit 0872604

**Pull request merged**